pkg/controllers/user/alert/manager/manager.go

```diff
@@ -0,0 +1,308 @@
+package manager
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/common/model"
+	alertconfig "github.com/rancher/rancher/pkg/controllers/user/alert/config"
+
+	"github.com/prometheus/alertmanager/dispatch"
+	"github.com/prometheus/alertmanager/types"
+	"github.com/rancher/types/apis/core/v1"
+	"github.com/rancher/types/config"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+type Manager struct {
+	nodeClient v1.NodeInterface
+	svcClient  v1.ServiceInterface
+	podClient  v1.PodInterface
+}
+
+func NewManager(cluster *config.UserContext) *Manager {
+	return &Manager{
+		nodeClient: cluster.Core.Nodes(""),
+		svcClient:  cluster.Core.Services(""),
+		podClient:  cluster.Core.Pods("cattle-alerting"),
+	}
+}
+
+//TODO: optimized this
+func (m *Manager) getAlertManagerEndpoint() string {
+
+	selector := labels.NewSelector()
+	r, _ := labels.NewRequirement("app", selection.Equals, []string{"alertmanager"})
+	selector.Add(*r)
+	pods, err := m.podClient.List(metav1.ListOptions{LabelSelector: selector.String()})
```

Comment:
any reason why cache is not used for listing pods?

Reply:
I will change to use Lister.List